program/src/processor.rs

```diff
@@ -519,68 +521,115 @@ pub fn process_update_exchange_rate(
     lido.save(accounts.lido)
 }
 
-/// If a stake account contains more inactive SOL than needed to make it rent-exempt,
-/// withdraw the excess back to the reserve. Returns how much SOL was withdrawn.
-pub fn withdraw_excess_inactive_sol<'a, 'b>(
-    accounts: &WithdrawInactiveStakeInfo<'a, 'b>,
-    clock: &Clock,
-    rent: &Rent,
-    stake_history: &StakeHistory,
-    stake_account: &AccountInfo<'b>,
+#[derive(PartialEq)]
+pub enum StakeType {
+    Stake,
+    Unstake,
+}
+
+impl std::fmt::Display for StakeType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            StakeType::Stake => write!(f, "stake"),
+            StakeType::Unstake => write!(f, "unstake"),
+        }
+    }
+}
+
+pub struct WithdrawExcessOpts<'a, 'b> {
+    accounts: &'a WithdrawInactiveStakeInfo<'a, 'b>,
+    clock: &'a Clock,
+    stake_history: &'a StakeHistory,
+    stake_account: &'a AccountInfo<'b>,
+}
+
+/// Withdraw `amount` from `withdraw_excess_opts.stake_account`.
+pub fn withdraw_inactive_sol(
+    withdraw_excess_opts: &WithdrawExcessOpts,
     stake_account_seed: u64,
```

Comment:
It looks like in all places where we use `WithdrawExcessOpts`, we also have a `stake_account_seed` argument. We could put the `stake_account_seed` in the `WithdrawExcessOpts`.